kernel/src/main/java/org/kframework/kompile/Kompile.java

```diff
@@ -297,9 +298,37 @@ private void checkDefinition(Definition parsedDef, Set<String> excludedModuleTag
         scala.collection.Set<Module> modules = parsedDef.modules();
         Module mainModule = parsedDef.mainModule();
         Option<Module> kModule = parsedDef.getModule("K");
+        definitionChecks(stream(modules).collect(Collectors.toSet()));
         structuralChecks(modules, mainModule, kModule, excludedModuleTags, true);
     }
 
+    // checks that are not verified in the prover
+    public void definitionChecks(Set<Module> modules) {
+        modules.forEach(m -> stream(m.localSentences()).forEach(s -> {
+            // Check that the `claim` keyword is not used in the definition.
+            if (s instanceof Claim)
+                errors.add(KEMException.compilerError("Claims are not allowed in the definition.", s));
+        }));
+    }
+
+    // Extra checks just for the prover specification.
+    public Module proverChecks(Module specModule, Module mainDefModule) {
+        // TODO: remove once transition to claim rules is done
+        // transform rules into claims if
+        // - they are in the spec modules but not in the definition modules
+        // - they don't contain the `simplification` attribute
+        ModuleTransformer mt = ModuleTransformer.fromSentenceTransformer((m, s) -> {
+            if (m.name().equals(mainDefModule.name()) || mainDefModule.importedModuleNames().contains(m.name()))
+                return s;
+            if (s instanceof Rule && !s.att().contains("simplification")) {
+                kem.registerCompilerWarning(KException.ExceptionType.FUTURE_ERROR, errors, "Deprecated: use claim instead of rule to specify proof objectives.", s);
+                return new Claim(((Rule) s).body(), ((Rule) s).requires(), ((Rule) s).ensures(), s.att());
+            }
+            return s;
+        }, "rules to claim");
+        return mt.apply(specModule);
+    }
```

Comment:
Add a test to `k-distribution/tests/regression-new/checkWarns` which exercises this case.

Reply:
It was a bit more complicated to modify the Makefile in that directory, so I made a new test in `regression-new` with a comment that it should be removed once the transition to claim is done.

Reply:
This is actually even more complicated than I originally thought. I modified `ktest.mak` to take into consideration stderr for the kprove tests. I modified all tests to use claim instead of the rule keyword, but there are other warnings inside those tests which were ignored until now, stuff like:
```
< [Warning] Compiler: Variable 'M' defined but not used. Prefix variable name
< with underscore if this is intentional.
< 	Source(./a1-spec.k)
< 	Location(12,19,12,20)
```
Or:
```
< kore-exec: [927827] Warning (WarnFunctionWithoutEvaluators):
<     (InfoReachability) while simplifying the claim:
<     No evaluators for function symbol:
<         Lbln'Unds'A'Unds'Int{}
```
^This last one I assume is from the haskell backend. I'm not sure I can fix this, and even if I fix it now, probably it's going to be unstable in the future.

So... thoughts?

Reply:
You can fix both of these I believe. The first one can eb fixed by changing `M => _M`. The second one can be fixed by changing the production `n_A_` to have the attribute `no-evaluators`.

But I don't think it should be necessary to catch all of `stderr`, should it? Maybe it is necessary.